lib/yard/registry_resolver.rb

```diff
@@ -185,5 +185,35 @@ def collect_namespaces(object)
 
       nss
     end
+
+    # @see NamespaceMapper#invalidate
+    def invalidate
```

Comment:
Unfortunately this won't trigger as `Namespacemapper.invalidate` is a module function. 

The good news is I think you can/should override `#clear_separators` instead, since NamespaceMapper's singleton stuff internal and only expected to be used _through_ / exposed by those instance methods.

```suggestion
    def clear_separators
```

(Need to change the docstring too)

Reply:
:+1: 

I now realize my previous comment wasn't true though: if `clear_separators` is renamed, this won't fail: it just silently won't be called. Do you want me to add a spec for that?

Reply:
Sure a spec would help.

Reply:
I just noticed that `invalidate` is also called via `register_separator`, so `register_separator` won't result in the memoizations being cleared, which seems necessary. So we need to extend `invalidate` after all? 

What we could do is add a line
```
invalidate if respond_to?(:invalidate)
```
to `register_separator`, after `NamespaceMapper.invalidate` so users of NamespaceMapper can define an instance method :invalidate for additional resets. What do you think about that?

Reply:
I opted to extend both #register_separator and #clear_separators